lib/fidor_api/model/base.rb

```diff
@@ -22,7 +25,7 @@ def resource_name
       attr_accessor :confirmable_action_id
 
       def as_json
-        attributes.compact
```

Comment:
@klausmeyer Thanks for the quick reply. I removed compact here, since it's only available in Ruby `>= 2.4`. Hope that is not a problem?
An alternative approach(from the top of my head) could be to have a monkey patch for Ruby `< 2.4`.

Reply:
I think we should keep the same behavior. We could use this instead:

```ruby
attributes.delete_if { |_, v| v.nil? }
```

Reply:
Much better. Added, thanks!

Reply:
Or

```ruby
attributes.reject { |_, v| v.nil? }
```

Reply:
Or

```ruby
attributes.reject { |_, v| v.nil? }
```

Reply:
Although, `delete_if` seems to be the faster one. Please let me know if we should keep it or change to `reject`

```ruby
2.5.1 :001 > require 'benchmark'
 => true
2.5.1 :002 >
2.5.1 :003 > iterations = 1_000_000
 => 1000000
2.5.1 :004 > attributes = { kaka: 'pupu', foo: nil }
 => {:kaka=>"pupu", :foo=>nil}
2.5.1 :005 >
2.5.1 :006 > Benchmark.bm do |bm|
2.5.1 :007 >       # using delete_if
2.5.1 :008 >       bm.report("delete_if:")  do
2.5.1 :009 >           iterations.times do
2.5.1 :010 >               attributes.delete_if { |_, v| v.nil? }
2.5.1 :011?>           end
2.5.1 :012?>       end
2.5.1 :013?>
2.5.1 :014?>     # using reject
2.5.1 :015?>     bm.report("reject:") do
2.5.1 :016 >           iterations.times do
2.5.1 :017 >               attributes.reject { |_, v| v.nil? }
2.5.1 :018?>           end
2.5.1 :019?>       end
2.5.1 :020?>   end
       user     system      total        real
delete_if:  0.141878   0.000698   0.142576 (  0.147592)
reject:     0.513890   0.006478   0.520368 (  0.524867)
```

Reply:
Think is that delete_if will directly operate on `attributes` (modify it) and `reject` returns a copy of the hash with the modifications. Might have side effects to change the `attributes`. That's why I still would prefer the `reject`.

Reply:
Agree, makes perfect sense in this case. Updated, thanks!